tests/phpunit/tests/block-bindings/postMetaSource.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Tests for Block Bindings API "core/post-meta" source.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.5.0
+ *
+ * @group blocks
+ * @group block-bindings
+ */
+class Tests_Block_Bindings_Post_Meta_Source extends WP_UnitTestCase {
+
+	/**
+	 * Sets up each test method.
+	 */
+	public function set_up() {
+		global $post;
+
+		parent::set_up();
+
+		$post = self::factory()->post->create_and_get( array() );
+		setup_postdata( $post );
+	}
+
+	/**
+	 * Tear down each test method.
+	 */
+	public function tear_down() {
+		// Removes custom fields registered by test cases.
+		$meta_keys = get_registered_meta_keys( 'post', '' );
+		foreach ( $meta_keys as $meta_key_name => $meta_key_value ) {
+			unregister_meta_key( 'post', $meta_key_name );
+		}
+
+		parent::tear_down();
+	}
+
+	/**
+	 * Tests that a block connected to a custom field renders its value.
+	 *
+	 * @ticket 60651
+	 */
+	public function test_custom_field_value_is_rendered() {
+		register_meta(
+			'post',
+			'custom_field',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+				'default'      => 'Custom field value',
+			)
+		);
+
+		$content = apply_filters( 'the_content', '<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"core/post-meta","args":{"key":"custom_field"}}}}} --><p>Fallback value</p><!-- /wp:paragraph -->' );
```

Comment:
This is what I see in other places:

```php
$parsed_blocks = parse_blocks( '<!-- wp:tests/context-consumer /-->' );
render_block( $parsed_blocks[0] );
```

I assume that using `the_content` filter also works but it's a bit magical when you see it for the first time.

Reply:
I tried using that approach first, but we need the post context to pass [this conditional](https://github.com/SantosGuillamot/wordpress-develop/blob/1736c839baa0e5cd40f077e8ad93474030733fa7/src/wp-includes/block-bindings/post-meta.php#L26-L28). If I just render a parsed block, there is no post ID.

Reply:
I see, so it somehow sets the post ID in the context during rendering this way.